src/interfaces.ts

```diff
@@ -106,93 +130,93 @@ interface RouteConfig {
   * You can optionally return a promise to tell the router to wait to bind and attach the view until
   * after you finish your work.
   */
-  activate: (params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction) => Promise<void>|PromiseLike<void>|IObservable|void;
+  activate(params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction): Promise<void> | PromiseLike<void> | IObservable | void;
 }
 
 /**
 * An optional interface describing the canDeactivate convention.
 */
-interface RoutableComponentCanDeactivate {
+export interface RoutableComponentCanDeactivate {
   /**
   * Implement this hook if you want to control whether or not the router can navigate away from your
   * view-model when moving to a new route. Return a boolean value, a promise for a boolean value,
   * or a navigation command.
   */
-  canDeactivate: () => boolean|Promise<boolean>|PromiseLike<boolean>|NavigationCommand;
+  canDeactivate: () => boolean | Promise<boolean> | PromiseLike<boolean> | NavigationCommand;
 }
 
 /**
 * An optional interface describing the deactivate convention.
 */
-interface RoutableComponentDeactivate {
+export interface RoutableComponentDeactivate {
   /**
   * Implement this hook if you want to perform custom logic when your view-model is being
   * navigated away from. You can optionally return a promise to tell the router to wait until
   * after you finish your work.
   */
-  deactivate: () => Promise<void>|PromiseLike<void>|IObservable|void;
+  deactivate: () => Promise<void> | PromiseLike<void> | IObservable | void;
 }
 
 /**
 * An optional interface describing the determineActivationStrategy convention.
 */
-interface RoutableComponentDetermineActivationStrategy {
+export interface RoutableComponentDetermineActivationStrategy {
   /**
   * Implement this hook if you want to give hints to the router about the activation strategy, when reusing
   * a view model for different routes. Available values are 'replace' and 'invoke-lifecycle'.
   */
-  determineActivationStrategy: (params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction) => 'no-change' | 'invoke-lifecycle'| 'replace';
+  determineActivationStrategy(params: any, routeConfig: RouteConfig, navigationInstruction: NavigationInstruction): 'no-change' | 'invoke-lifecycle' | 'replace';
 }
 
 /**
 * An optional interface describing the router configuration convention.
 */
-interface ConfiguresRouter {
+export interface ConfiguresRouter {
   /**
   * Implement this hook if you want to configure a router.
   */
-  configureRouter(config: RouterConfiguration, router: Router): Promise<void>|PromiseLike<void>|void;
+  configureRouter(config: RouterConfiguration, router: Router): Promise<void> | PromiseLike<void> | void;
 }
 
 /**
 * An optional interface describing the available activation strategies.
 */
-interface ActivationStrategy {
+export interface ActivationStrategy {
```

Comment:
Should we convert this to an enum?

Reply:
I was doing so, that's why you see `,` instead of `;` in its fields. I'm not sure about it, because it's already exported, and no1 uses this interface, only the `export const activationStrategy`

Reply:
This interface never made any sense to begin with because it describes an object containing activationStrategies, not an actual activationStrategy (which is just a plain string).
*This* would make sense: `export type ActivationStrategy = 'no-change' | 'invoke-lifecycle' | 'replace'`

What this interface *could* be useful as, is `HasActivationStrategy`. E.g. you could make a ViewModel implement it. But then still it doesn't make sense because it doesn't have a property `activationStrategy: 'no-change' | 'invoke-lifecycle' | 'replace'` (like it should).

Since the object (or enum, they're the same thing) are both capable of describing themselves, this interface is unquestionably redundant.

Reply:
@fkleuver Actually the `interface ActivationStrategy` should describe a `const` similar to the `bindingMode` one. Though not sure where it is 😅

Reply:
@StrahilKazlachev here https://github.com/aurelia/router/blob/master/src/navigation-plan.js#L6

@bigopon Sounds like we can remove this completely, right?

Reply:
Yes, probably. I added a dedicated interface
https://github.com/aurelia/router/blob/a0135eba04cdf222795fff2c58cedd6a1987969e/src/interfaces.ts#L210

Reply:
@bigopon Should this be removed?